test/source/mock/attester/attester-endpoints.ts

```diff
@@ -1,437 +1,88 @@
-      }
-      if (emailOrLongid === 'test.flowcrypt.pubkeyserver.priority@gmail.com') {
-        return protonMailCompatKey;
-      }
-      if (emailOrLongid === 'has.pub@client-configuration-test.flowcrypt.test') {
-        return hasPubKey;
-      }
-      if (emailOrLongid === 'invalid.pub@client-configuration-test.flowcrypt.test') {
-        return protonMailCompatKey;
-      }
-      if (emailOrLongid === 'test.ldap.keyserver.pgp@gmail.com' && server === 'keyserver.pgp.com') {
-        return [protonMailCompatKey, testMatchPubKey].join('\n');
-      }
-      if (emailOrLongid === 'test.ldap.timeout@gmail.com') {
-        throw new HttpClientErr('RequestTimeout', Status.BAD_REQUEST);
-      }
-      if (emailOrLongid === 'test.flowcrypt.pubkey.timeout@gmail.com') {
-        return somePubkey;
-      }
-      throw new HttpClientErr('No OpenPGP LDAP server on this address.', Status.NOT_FOUND);
-    } else {
-      throw new HttpClientErr(`Not implemented: ${req.method}`);
-    }
-  },
-  '/attester/welcome-message': async ({ body }, req) => {
-    if (!isPost(req)) {
-      throw new HttpClientErr(`Wrong method: ${req.method}`);
-    }
-    const { email, pubkey } = body as Dict<string>;
-    expect(email).to.contain('@');
-    expect(pubkey).to.contain('-----BEGIN PGP PUBLIC KEY BLOCK-----');
-    return { sent: true };
-  },
+interface PubKeyLookUpResult {
+  pubkey?: string;
+  delayInSeconds?: number;
+  domainToCheck?: string;
+  returnError?: { code: number; message: string };
+}
+
+export interface AttesterConfig {
+  pubkeyLookup?: Record<string, PubKeyLookUpResult>;
+  ldapRelay?: Record<string, PubKeyLookUpResult>;
+  welcomeMessageEnabled?: boolean;
+}
+
+export const getMockAttesterEndpoints = (attesterConfig: AttesterConfig): HandlersDefinition => {
+  return {
+    '/attester/pub/?': async ({ body }, req) => {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const emailOrLongid = req.url!.split('/').pop()!.toLowerCase().trim();
+      if (isGet(req)) {
+        if (!attesterConfig?.pubkeyLookup) {
+          throw new HttpClientErr('Method not allowed', 405);
+        }
+        const pubRes = attesterConfig.pubkeyLookup[emailOrLongid];
+        if (pubRes) {
+          if (pubRes.returnError) {
+            throw new HttpClientErr(pubRes.returnError.message, pubRes.returnError.code);
+          }
+          if (pubRes.delayInSeconds) {
+            await Util.sleep(pubRes.delayInSeconds);
+          }
+          return pubRes.pubkey;
+        }
+        throw new HttpClientErr('Pubkey not found', 404);
+      } else if (isPost(req)) {
+        oauth.checkAuthorizationForEmail(req.headers.authorization, emailOrLongid);
+        expect(body).to.contain('-----BEGIN PGP PUBLIC KEY BLOCK-----');
+        if (attesterConfig.pubkeyLookup) {
+          attesterConfig.pubkeyLookup[emailOrLongid] = { pubkey: body as string };
+          return 'Saved'; // 200 OK
+        }
```

Comment:
probably it's possible that `attesterConfig.pubkeyLookup` will be `null`?
then we need `else`

```ts
if (attesterConfig.pubkeyLookup) {
   attesterConfig.pubkeyLookup[emailOrLongid] = { pubkey: body as string };
} else {
   attesterConfig.pubkeyLookup = { emailOrLongid: { pubkey: body as string }};
}
return 'Saved'; // 200 OK
```